File: pset2/vigenere/vigenere.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>

int main(int argc, string argv[])
{
    //checking number of arguments
    if (argc != 2)
    {
        printf("Wrong Number Of Arguments\n");
        return 1;
    }
    //checking that the entered string has all characters as alphabets
    //since the entered argument is a string, so have to check each character one by one
    //putting the entered argument in a single string to make it 1D from 2D matrix
    string k = argv[1], pt;
    int sizek = strlen(k);
    for (int i = 0; i < sizek; i++)
    {
        if (!isalpha(k[i]))
        {
            return 1;
        }
        if (islower(k[i]))
        {
            k[i] = (int) k[i] - 97;
        }
        if (isupper(k[i]))
        {
            k[i] = (int) k[i] - 65;
        }
    }

    pt = get_string("plaintext: ");
    string ct = pt;
    int size = strlen(pt);
    for (int j = 0, c = 0; j < size; j++)
    {
        if (c < sizek)
        {
            if (islower(pt[j]))
            {
                ct[j] = (((pt[j] + k[c]) - 97) % 26) + 97;
                c++;
            }
            if (isupper(pt[j]))
            {
                ct[j] = (((pt[j] + k[c]) - 65) % 26) + 65;
                c++;

            }

        }
        if (c == sizek)
        {
            c = 0;
        }
    }
    printf("ciphertext: %s\n", ct);
}

File: pset6/sentimental/vigenere/vigenere.py
import sys
from cs50 import get_string


def main():
    if len(sys.argv) != 2:
        print('Wrong number of arguments')
        sys.exit(1)
    k = sys.argv[1]
    sizek = len(k)
    code = list()
    for i in k:
        if not i.isalpha():
            sys.exit(1)
        if i.islower():
            code.append(ord(i) - 97)
        if i.isupper():
            code.append(ord(i) - 65)

    pt = get_string('Plaintext: ')
    ct = list()
    size = len(pt)
    c = 0
    for j in range(0, size):
        if c < sizek:
            if pt[j].islower():
                ct.append(chr((((ord(pt[j]) + code[c]) - 97) % 26) + 97))
                c = c + 1
            elif pt[j].isupper():
                ct.append(chr((((ord(pt[j]) + code[c]) - 65) % 26) + 65))
                c = c + 1
            else:
                ct.append(pt[j])
        if c == sizek:
            c = 0
    print('ciphertext:', ''.join(str(e) for e in ct))


if __name__ == "__main__":
    main()